### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by bobb-Rob**
I implemented the following: 

* Create a class Classroom with the following:
    - @label instance variable, should be initialized in the constructor.
    - Setter and getter for @label (remember about attr_accessor).

* Create the has-many/belongs-to relationship between Classroom and Student. The following should be implemented:

- Create the has-many side (a classroom has many students).
- Create the belongs-to side (a student belongs to a classroom).

- Create a class Book with the following:
     - @title and @author instance variables, should be initialized in the constructor.
    - Setters and getters for instance variables (remember about attr_accessor).

- Create a class Rental with the following:
     - @date instance variable, should be initialized in the constructor.
     - Setter and getter for @date (remember about attr_accessor).

- Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental. The following should be implemented:

     - Create the has-many side of Book and Rental (a book has many rentals).
     - Create the belongs-to side of Rental and Book (a rental belongs to a book).
     - Create the has-many side of Person and Rental (a person has many rentals).
     - Create the belongs-to side of Rental and Person (a rental belongs to a person).
     - Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classes/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Classroom
+  attr_accessor :label
+  attr_reader :classroom
+
```
Comment (RolandM99) [inline classes/classroom.rb]: - [x] Kindly, make sure that you define the relationship between Classroom and Student. `attr_reader :students`

File: classes/student.rb [hunk 2]
```diff
@@ -1,7 +1,9 @@
 require_relative 'person'
 
 class Student < Person
-  def initialize(classroom, age, name = 'unknown', parent_permission: true)
+  attr_accessor :classroom
+
+  def initialize(age, name = 'unknown', classroom = 'unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
     @classroom = classroom
   end
```
Comment (RolandM99) [inline classes/student.rb:9]: - [x] Kindly, make sure that you implement a method that serves as the `belongs_to` relationship between Classroom and Student as student belongs to classroom. have a look on the code below:
```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

File: classes/book.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (RolandM99) [inline classes/book.rb]: - [x] Make sure that you create a method for adding rentals in the Book class. The book itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Check the example below:
```ruby
def add_rental(person, date)
    Rental.new(date, self, person)
end
```

File: classes/person.rb [hunk 4]
```diff
@@ -3,18 +3,21 @@
...
   def initialize(age, name, parent_permission: true)
     @id = rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
     super()
   end
 
   attr_reader :id
```
Comment (RolandM99) [inline classes/person.rb:19]: - [x] Make sure that you create a method for adding rentals in the Person class. The person itself should be sent as a parameter to create the new rental achieving the 'has-many' association. Check the example below:
```ruby
def add_rental(book, date)
    Rental.new(date, self, book)
end
```

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hi  @bobb-Rob, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Gitflow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: classes/classroom.rb (hunk 1 above)
Comment (bobb-Rob) [reply classes/classroom.rb]: @RolandM99 , 
Thank you for the invaluable review comments. But I don't understand what this means in a practical setting for me. I am doing my best to understand this, and a little more explanation can help.

Comment (bobb-Rob) [reply classes/classroom.rb]: @RolandM99 I think I you mean I use this `attr_accessor :students` instead of `attr_reader :classroom`

Comment (RolandM99) [reply classes/classroom.rb]: What I'm trying to tell you is just simple,  replacing `:classroom` with `students`. Actually in your code, it's kinda like you're defining getter for classroom ignoring the relationship with student.
So, just replace `:classroom` with `:students`

**Review by zainsadaqat — APPROVED** [review]
# APPROVED!!!! 🎉 🎉 🎉 🎉 🎉 🎉
## Hey @bobb-Rob  😊 😃 

#### Great Job 😄 👏 You've done it. Your project is complete!  There is nothing else to say other than... it's time to merge it 👍🏻 ✅ 

# Congratulations!!! 🎉 🥳 👏 🙌 🎊 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**